### PR TITLE
Jgfouca/branch for acme split 2018 06 25

### PR DESCRIPTION
**Pull request by jgfouca**
Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: [bit for bit, roundoff, climate changing]

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review: @jedwards4b

### CODE REVIEW

File: scripts/lib/CIME/SystemTests/homme.py [hunk 1]
```diff
@@ -27,14 +27,19 @@ def build_phase(self, sharedlib_only=False, model_only=False):
...
-            basegen  = self._case.get_value("BASEGEN_CASE")
             basecmp  = self._case.get_value("BASECMP_CASE")
-            generate = self._case.get_value("GENERATE_BASELINE")
+            compare  = self._case.get_value("COMPARE_BASELINE")
             gmake    = self._case.get_value("GMAKE")
             cprnc    = self._case.get_value("CCSM_CPRNC")
 
-            basename = basegen if generate else basecmp
-            cmake_cmd = "cmake -C {}/components/homme/cmake/machineFiles/{}.cmake -DUSE_NUM_PROCS={} {}/components/homme -DHOMME_BASELINE_DIR={}/{} -DCPRNC_DIR={}/..".format(srcroot, mach, procs, srcroot, baseline, basename, cprnc)
+            if compare:
```
Comment (jgfouca) [inline scripts/lib/CIME/SystemTests/homme.py]: This is a real change, but it won't impact CESM.

File: scripts/lib/CIME/case/case_run.py [hunk 2]
```diff
@@ -110,9 +113,15 @@ def _run_model_impl(case, lid, skip_pnl=False, da_cycle=0):
     while loop:
         loop = False
 
+        logger.info("{} SAVE_PRERUN_PROVENANCE BEGINS HERE".format(time.strftime("%Y-%m-%d %H:%M:%S")))
```
Comment (jgfouca) [inline scripts/lib/CIME/case/case_run.py]: Real change, lots of additional time logs.

**Comment by mfdeakin-sandia** [comment]
I guess this means I should get the LSF fix and the test for the arguments to case.submit done in the next few days

File: scripts/lib/CIME/utils.py [hunk 3]
```diff
@@ -427,12 +427,12 @@ def run_cmd(cmd, input_str=None, from_dir=None, verbose=None,
     output, errput = proc.communicate(input_str)
     if output is not None:
         try:
-            output = output.decode('utf-8').strip()
+            output = output.decode('utf-8', errors='ignore').strip()
```
Comment (jgfouca) [inline scripts/lib/CIME/utils.py]: Real change.

File: src/drivers/mct/cime_config/config_component_e3sm.xml [hunk 4]
```diff
@@ -172,22 +172,26 @@
...
       <value compset="HIST.*_DATM%(QIA|CRU|GSW)">CO2A</value>
       <value compset="20TR.*_DATM%(QIA|CRU|GSW)">CO2A</value>
       <value compset="RCP.*_DATM%(QIA|CRU|GSW)" >CO2A</value>
       <value compset="_DATM%IAF.*_MPASO%OECO" >CO2A</value>
       <value compset="_MPASCICE%BGC.*_MPASO%OIECO" >CO2A_OI</value>
       <value compset="_MPASCICE%BGC.*_MPASO%TOIECO_" >CO2A_OI</value>
-      <value compset="_BGC%BCRD">CO2C</value>
-      <value compset="_BGC%BDRC">CO2C</value>
-      <value compset="_BGC%BCRC">CO2C</value>
+      <value compset="_MPASCICE_MPASO_.*_BGC%BCRC" >CO2C</value>
```
Comment (jgfouca) [inline src/drivers/mct/cime_config/config_component_e3sm.xml]: Real change.

File: scripts/lib/CIME/provenance.py [hunk 5]
```diff
@@ -362,8 +362,8 @@ def _save_postrun_timing_e3sm(case, lid):
             globs_to_copy.append("%s*run*%s" % (case.get_value("CASE"), job_id))
 
     globs_to_copy.append("logs/run_environment.txt.{}".format(lid))
-    globs_to_copy.append("logs/e3sm.log.{}.gz".format(lid))
-    globs_to_copy.append("logs/cpl.log.{}.gz".format(lid))
+    globs_to_copy.append(os.path.join(rundir, "e3sm.log.{}.gz".format(lid)))
```
Comment (jgfouca) [inline scripts/lib/CIME/provenance.py]: No impact to cesm.

File: scripts/lib/CIME/case/case_run.py [hunk 6]
```diff
@@ -62,12 +62,15 @@ def _pre_run_check(case, lid, skip_pnl=False, da_cycle=0):
     # This needs to be done everytime the LID changes in order for log files to be set up correctly
     # The following also needs to be called in case a user changes a user_nl_xxx file OR an env_run.xml
     # variable while the job is in the queue
+    logger.info("{} NAMELIST CREATION BEGINS HERE".format(time.strftime("%Y-%m-%d %H:%M:%S")))
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply scripts/lib/CIME/case/case_run.py]: You mean demoted?

**Comment by jgfouca** [comment]
@jedwards4b , i changed to debug.

File: config/e3sm/machines/config_machines.xml [hunk 7]
```diff
@@ -2911,11 +2870,11 @@
 		<command name="load">netlib-lapack/3.6.1</command>
       </modules>
       <!-- List of modules elements, executing commands if compiler and mpilib condition applies -->
-      <modules compiler="pgi">
+      <modules compiler="pgi.*">
```
Comment (jedwards4b) [inline config/e3sm/machines/config_machines.xml]: This seems strange - I didnt thing regular expressions would work here.

**Pull request merged**